### ISSUE TITLE
Consider enacting some sort consent requirement?

### ISSUE THREAD

**Issue by Casikao** [jaycewhite/MiniMappingway#27]
Currently; this plugin removes all consent to location for someone as a one-way avenue. Most (Good Friends, etc.) Require a pairing for people to install the plugin in order for it to communicate and work.

With friends also being a 1 way street in XIV, this means if someone is removed from a friends list, or blacklisted; they do not appear on my friends list, but I would still appear on theirs, which then grants them pinpoint access to where I am on a map at any time.

It'd be nice to have this have a limit of opting in. I..only know this is the case due to someone who has used this to do the near-exact case as above.

**Comment by jaycewhite** [comment]
Hi! 

Firstly, thanks for the feedback!

So this has been raised to me a couple of times recently, and I have given it much thought. I believe the amount of time/effort required to implement such a pairing system is fairly prohibitive compared to the benefit of said system. here is my reasoning:

1. This would require me to write server architecture to handle the pairing/"friending" of people, and would require me to host said system myself which could be costly. This would easily (imo) take more time than it took for me to write the entirety of the rest of the plugin, for very little visible benefit (and actually some added awkwardness) for most users.

2. Currently, the plugin does not show you anything you cannot already see in game. The way the plugin works is to scan players who are actively in your game world (so fairly close by, depending on player density in your area). This means that any time anyone can use this plugin to find you, there are also myriad other ways to do so. Other plugins can hide everyone except friends for example, and just in the vanilla game itself, nameplates will show a different colour for friends, and the /target <player name> command can be used to target anyone in game also. The plugin doesn't display any information not already available to the player, it simply adds another way to view that data.

3. The friends list working the way it does is... perplexing to say the least, but just being able to see that person on your friends list/player search is already pretty problematic in these instances. Ideally the friends list would just remove the other person when you unfriend them, and working around this system as currently implemented would make for a bad user experience. It would require everyone using the plugin to connect to a server and manually add all of their friends/fc-mates, just to display _less_ information than the game itself already gives you, and imo it would compound onto an already awkward friend system, which isn't really desirable.

I've rambled a bit, but in summary, the plugin doesn't show you anything the game doesn't already; it would be a lot of time/effort spent on mitigating the problem, and it would make for a much worse user experience (+ a much higher barrier for entry, not to mention never being able to see console friends on the minimap). 

I do hear you though, I have made decisions in the past to try and limit stalking etc, and it is something I do bear in mind. For example: It has been requested a few times to let people manually add names to the plugin for it to track. This was actually requested for the opposite case of avoiding people that are actively trying to stalk you, but I declined to make this feature due to the stalking concern.

Hope this all makes sense!
